### PR TITLE
:sparkles: New UI for grid layout

### PR DESCRIPTION
**Pull request by Alotor**
> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs [hunk 1]
```diff
@@ -832,13 +870,18 @@
              (st/emit! (udw/start-edition-mode id))
              (st/emit! :interrupt))))]
     (if new-css-system
-      [:div "new-edit-mode"]
+      [:button
+       {:class (stl/css :edit-mode-btn)
+        :alt  "Grid edit mode"
+        :on-click #(toggle-edit-mode)}
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: Avoid using inline functions on "on-click" better define them on let.

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs [hunk 2]
```diff
@@ -926,74 +983,139 @@
...
+           :options [{:value :flex :label "FR"}
+                     {:value :auto :label "AUTO"}
+                     {:value :fixed :label "PX"}
+                     {:value :percent :label "%"}]
+           :on-change #(set-column-type type index %)}]]]
+
+       [:button
+        {:class (stl/css :remove-track-btn)
+         :on-click #(remove-element type index)}
+        i/minus]]
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: Use new icon i/remove-refactor

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs [hunk 3]
```diff
@@ -926,74 +983,139 @@
...
-                                :hover-track hover-track}])]])]))
+        add-track
+        #(do
+           (when-not expanded? (toggle))
+           (add-new-element type ctl/default-track-value))]
+
+    (if new-css-system
+      [:div {:class (stl/css :grid-tracks)}
+       [:div {:class (stl/css :grid-track-header)}
+        [:button {:class (stl/css :expand-icon) :on-click toggle} i/actions]
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: Use new icon i/menu-refactor

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs [hunk 4]
```diff
@@ -926,74 +983,139 @@
...
+        [:div {:class (stl/css :track-title) :on-click toggle}
+         [:div {:class (stl/css :track-name) :title track-name} track-name]
+         [:div {:class (stl/css :track-detail) :title track-detail} track-detail]]
+        [:button {:class (stl/css :add-column) :on-click add-track} i/plus]]
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: Use new icon i/add-refactor

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss [hunk 5]
```diff
@@ -63,65 +63,220 @@
...
+    @include flexColumn;
+    margin-top: $s-8;
+  }
+}
+
+.track-info {
+  display: flex;
+
+  &.dnd-over-top {
+    border-top: 2px solid var(--button-foreground-hover);
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: change px to variable

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss [hunk 6]
```diff
@@ -63,65 +63,220 @@
...
+  }
+
+  &.dnd-over-bot {
+    border-bottom: 2px solid var(--button-foreground-hover);
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: change px to variable

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss [hunk 7]
```diff
@@ -63,65 +63,220 @@
...
+      @extend .button-icon;
+      stroke: var(--icon-foreground);
+      height: 100%;
+    }
+  }
+
+  .track-info-value {
+    @extend .input-element;
+    border-radius: 0;
+    border-right: 1px solid var(--panel-background-color);
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: change px to variable

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss [hunk 8]
```diff
@@ -63,65 +63,220 @@
...
+
+  .track-name {
+    color: var(--color-foreground-secondary);
+  }
+
+  .expand-icon {
+    @extend .button-secondary;
+    height: $s-52;
+
+    border-radius: calc(var(--s-4) * 2) 0 0 calc(var(--s-4) * 2);
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: change to $br-8 ?

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss [hunk 9]
```diff
@@ -63,65 +63,220 @@
...
+    border-right: 1px solid var(--panel-background-color);
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: change px to variable

File: frontend/src/app/main/ui/workspace/viewport/grid_layout_editor.cljs [hunk 10]
```diff
@@ -656,22 +739,22 @@
...
+            :class (stl/css :grid-editor-label)
             :type "text"
             :default-value (format-size track-data)
             :data-default-value (format-size track-data)
             :on-key-down handle-keydown-track-input
             :on-blur handle-blur-track-input}]
           (when (and hovering? (not medium?) (not small?))
-            [:button {:class (css :grid-editor-button)
+            [:button {:class (stl/css :grid-editor-button)
                       :on-click handle-remove-track} i/trash])]])]
```
Comment (EvaMarco) [inline frontend/src/app/main/ui/workspace/viewport/grid_layout_editor.cljs]: chnage to new icon

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs (hunk 1 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs (hunk 2 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs (hunk 3 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs (hunk 4 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss (hunk 5 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss (hunk 7 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: ok

File: frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss (hunk 8 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.scss]: ok

**Pull request merged**